addons/account/models/account_move.py

```diff
@@ -2879,12 +2887,23 @@ def _onchange_credit(self):
             self.debit = 0.0
         self._onchange_balance()
 
-    @api.onchange('amount_currency')
+    @api.onchange('amount_currency', 'currency_id')
```

Comment:
You need to manage that in `_onchange_currency`. You can't add this dependency there.